src/business-logic/services/test/device-service.spec.ts

```diff
@@ -9,7 +9,7 @@ describe('DeviceService', () => {
   let testDevice: TelemetricsDevice
   let deviceRepository: MongoDeviceRepository
   let deviceEventEmitter: DeviceEventEmitter
-  const mockedValue = mock<MongoDeviceRepository>()
+  const mockedValue: MongoDeviceRepository = mock<MongoDeviceRepository>()
```

Comment:
This is still using `MongoDeviceRepository` which is a thing only the data-access-layer may know about.
You are in the business-logic-layer so you should use the abstraction `DeviceRepository` instead of a concrete implementation.